Building package before codecgen generation breaks if types changed

#1042 builds a package before invoking codecgen to workaround https://github.com/ugorji/go/issues/145

However, as @foot experienced, that's not enough: building the package will fail if the types changed since the generated codec will be outdated.

```
env -u GOARCH -u GOOS env GO15VENDOREXPERIMENT=1 GOGC=off go build -i -tags 'netgo unsafe' ./render/detailed
# github.com/weaveworks/scope/render
render/render.codecgen.go:56: undefined: report.NodeSet
render/render.codecgen.go:1525: undefined: report.NodeSet
render/render.codecgen.go:1765: undefined: report.NodeSet
Makefile:82: recipe for target 'render/detailed/detailed.codecgen.go' failed
make: *** [render/detailed/detailed.codecgen.go] Error 2
make: Leaving directory '/go/src/github.com/weaveworks/scope'
make: *** [render/detailed/detailed.codecgen.go] Error 2
```

Full logs: https://gist.github.com/foot/b0e70c7563b0fd2b80a5

--- a/Makefile
+++ b/Makefile
@@ -79,7 +79,7 @@ $(SCOPE_EXE): $(SCOPE_BACKEND_BUILD_UPTODATE)
 	    }
 
 %.codecgen.go: $(SCOPE_BACKEND_BUILD_UPTODATE)
-	env -u GOARCH -u GOOS $(GO) build -i -tags $(GO_BUILD_TAGS) ./$(@D)
+	rm -f $@ && env -u GOARCH -u GOOS $(GO) build -i -tags $(GO_BUILD_TAGS) ./$(@D) # workaround for https://github.com/ugorji/go/issues/145
 	cd $(@D) && env -u GOARCH -u GOOS GOGC=off $(shell pwd)/$(CODECGEN_EXE) -rt $(GO_BUILD_TAGS) -u -o $(@F) $(notdir $(call GET_CODECGEN_DEPS,$(@D)))
 
 $(CODECGEN_EXE): $(SCOPE_BACKEND_BUILD_UPTODATE)

